autogen/agentchat/conversable_agent.py

```diff
@@ -731,6 +738,148 @@ async def a_initiate_chat(
+                        when set "reflection_with_llm", it returns the takeaway extracted using an llm client.
+                        "reflection_with_llm" requires the llm_config to be set in either the sender or the recipient.
+                    "takeaway_prompt" can be used to specify the prompt used to extract the takeaway when takeaway_method is "reflection_with_llm".
+                        Default is None and the following default prompt will be used when "takeaway_method" is set to "reflection_with_llm":
+                        "Identify and extract the final solution to the originally asked question based on the conversation."
+                    "carryover" can be used to specify the carryover information to be passed to this chat.
+                        If provided, we will combine this carryover with the "message" content when generating the initial chat
+                        message in `generate_init_message`. Please override the `generate_init_message` if you want to customize
+                        how to use the carryover information shall be used.
+
+        """
+        receipts_set = set()
+        for chat_info in chat_queue:
+            assert "recipient" in chat_info, "recipient must be provided."
+            receipts_set.add(chat_info["recipient"])
+        assert len(receipts_set) == len(chat_queue), "recipients must be different."
+
+        self._chat_queue = chat_queue
+        self._finished_chats = {}
+        takeaway = []
+        while self._chat_queue:
+            chat_info = self._chat_queue.pop(0)
+            _chat_carryover = chat_info.get("carryover", [])
+            if isinstance(_chat_carryover, str):
+                _chat_carryover = [_chat_carryover]
+            chat_info["carryover"] = _chat_carryover + list(self._finished_chats.values())
+            if "message" not in chat_info:
+                warnings.warn(
+                    "message is not provided in chat_info. input() will be called to get the initial message.",
+                    UserWarning,
+                )
+            current_agent = chat_info["recipient"]
+            print_carryover = (
+                ("\n").join([t for t in chat_info["carryover"]])
+                if isinstance(chat_info["carryover"], list)
+                else chat_info["carryover"]
+            )
+            print(colored("\n" + "*" * 80, "blue"), flush=True, sep="")
+            print(
+                colored(
+                    "Start work on the following task: \n"
+                    + chat_info.get("message")
+                    + "\n\nWith the following carryover from previous tasks: \n"
+                    + print_carryover,
+                    "blue",
+                ),
+                flush=True,
+            )
+            print(colored("\n" + "*" * 80, "blue"), flush=True, sep="")
+            takeaway = self.initiate_chat(**chat_info)
+
+            self._finished_chats[current_agent] = takeaway
```

Comment:
I like that there is a variable that stores the results of each finished sub task. 

As a developer, a common use case is that I want some final output.
This may be   the final result from all of the tasks or the final result from the last task. What is the way to do this? E.g. agent._finished_chats[last_agent][-1].
It would be good to have an example in the notebook

Reply:
Added an example in the notebook (code block [6]). Is it what you want to see?

Reply:
Right. A `ChatResult` object could be useful here, as in my earlier comments.

Reply:
Added.